MantidPlot/src/Mantid/InstrumentWidget/PeakOverlay.h

```diff
@@ -28,50 +28,110 @@ class UnwrappedSurface;
  * A label shows three numbers h,k, and l. A combined label replaces non-equal
  * numbers of included markers with its letter.
  */
-class PeakHKL
-{
+class PeakHKL {
 public:
-  PeakHKL(PeakMarker2D* m,const QRectF& trect,bool sr);
-  bool add(PeakMarker2D* marker,const QRectF& trect);
-  void draw(QPainter& painter,int prec = 6);
-  void print()const;
+  PeakHKL(PeakMarker2D *m, const QRectF &trect, bool sr);
+  bool add(PeakMarker2D *marker, const QRectF &trect);
+  void draw(QPainter &painter, int prec = 6);
+  void print() const;
 
 private:
   static QString formatNumber(double h, int prec);
-  QPointF p; ///< untransformed marker origin
-  QRectF rect; ///< label's screen area in transformed coords
-  double h,k,l; ///< h,k, and l
-  bool nh,nk,nl; ///< true if h, k, or l is numeric
+  QPointF p;       ///< untransformed marker origin
+  QRectF rect;     ///< label's screen area in transformed coords
+  double h, k, l;  ///< h,k, and l
+  bool nh, nk, nl; ///< true if h, k, or l is numeric
   QList<int> rows; ///< row indices of the peaks in their PeaksWorkspace
   bool showRows;
 };
 
+/// Helper class for scaling peak markers to intensities.
+class AbstractIntensityScale {
+public:
+  AbstractIntensityScale(
+      const boost::shared_ptr<Mantid::API::IPeaksWorkspace> &pws) {
+    setPeaksWorkspace(pws);
+  }
```

Comment:
Shouldn't there be a virtual destructor? You have pointers to child classes, I think.